hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MulticastJoiner.java

```diff
@@ -148,7 +141,7 @@ public void onMessage(Object msg) {
                 logger.warning(e);
             }
         } finally {
-            node.multicastService.removeMulticastListener(listener);
+            splitBrainJoinMessages.clear();
```

Comment:
As I understand, having `splitBrainJoinMessages.clear()` at the end of each cycle, join messages will be alive at most one cycle of split discovery.